external/storm-kafka/src/jvm/storm/kafka/PartitionManager.java

```diff
@@ -137,8 +137,14 @@ public EmitState next(SpoutOutputCollector collector) {
             }
             Iterable<List<Object>> tups = KafkaUtils.generateTuples(_spoutConfig, toEmit.msg);
             if (tups != null) {
-                for (List<Object> tup : tups) {
-                    collector.emit(tup, new KafkaMessageId(_partition, toEmit.offset));
+		if(_spoutConfig.topicAsStreamId) {
+	            for (List<Object> tup : tups) {
```

Comment:
@harshach 
Are you sure moving the if inside the loop will not be a performance overhead ?
And about renaming the variables, I'm not sure if I should be changing variable names I did not create.